CorsixTH/Src/main.cpp

```diff
@@ -83,28 +167,18 @@ int lua_main_no_eval(lua_State* L) {
   // require "debug" (Harmless in Lua 5.1, useful in 5.2 for compatibility)
   luaT_execute(L, "require \"debug\"");
 
-  // Check for --interpreter and run that instead of CorsixTH.lua
-  bool bGotScriptFile = false;
-  int iNArgs = lua_gettop(L);
-  for (int i = 1; i <= iNArgs; ++i) {
-    if (lua_type(L, i) == LUA_TSTRING) {
-      size_t iLen;
-      const char* sCmd = lua_tolstring(L, i, &iLen);
-      if (iLen > 14 && std::memcmp(sCmd, "--interpreter=", 14) == 0) {
-        lua_getglobal(L, "assert");
-        lua_getglobal(L, "loadfile");
-        lua_pushlstring(L, sCmd + 14, iLen - 14);
-        bGotScriptFile = true;
-        break;
-      }
-    }
-  }
+  lua_getglobal(L, "assert");
+  lua_getglobal(L, "loadfile");
 
-  if (!bGotScriptFile) {
-    lua_getglobal(L, "assert");
-    lua_getglobal(L, "loadfile");
-    lua_pushstring(L, CORSIX_TH_INTERPRETER_PATH);
+  auto scriptFilePath = search_script_file(L);
+  if (scriptFilePath.empty()) {
+    std::fprintf(stderr,
+                 "CorsixTH cannot find CorsixTH.lua. If you want use a custom "
+                 "location, specify it by --interpreter=FILE\n");
+    std::fflush(stderr);
+    exit(1);
   }
+  lua_pushstring(L, scriptFilePath.c_str());
```

Comment:
The lua c API all translates to `assert(loadfile(scriptFIlePath))`  which is easier to read when it's in one spot. Move the lua_getglobal lines down with the lua_pushstring.